src/test/java/nl/novi/PersonTest.java

```diff
@@ -0,0 +1,233 @@
+
+    @Test
+    void getFatherTest() {
+        // Arrange
+        Person person1 = new Person("Marianne", "Bosma", 'V', 40);
+        Person person2 = new Person("Christiaan", "Frederik", "Bosma", 'M', 68);
+        person1.setFather(person2); // De setterfunctie wordt meteen meegetest
+        // Act
+        Person father = person1.getFather();
+        // Assert
+        assertEquals(person2, father);
+    }
+
+
+    @Test
+    void getChildrenTest() {
+        // Arrange
+        Person person1 = new Person("Marjet", "Johanna","Bosma", 'V', 42);
+        Person person2 = new Person("Hannah", "Esther", "Daalder", 'V', 10);
+        List<Person> children = new ArrayList<>();
+        children.add(person2);
+        // Act
+        List<Person> mySistersChildren = person1.getChildren();
+        // Assert
+        assertEquals(children, mySistersChildren);
+    }
+
+    @Test
+    void getSiblingsTest() {
+        // Arrange
+        Person person1 = new Person("Marjet", "Johanna", "Bosma", 'V', 42);
+        Person person2 = new Person("Marianne", "Bosma", 'V', 40);
+        List<Person> children = new ArrayList<>(); // zussen worden in
+        children.add(person2);
+        person1.setSiblings(children); // setterfunctie meegetest
+        // Act
+        List<Person> siblings = person1.getSiblings();
+        // Assert
+        assertEquals(children, siblings);
+    }
+
+    @Test
+    void addChildTest() {
+        father.addChild(father, me);
+        father.addChild(father, sister);
+        assertEquals("Marjet", father.getChildren().get(0).getName()); // get(0): eerste index van de lijst
+        assertEquals(2, father.getChildren().size());
+    }
+
+    @Test
+    void addSiblingTest() {
+        me.addSibling(me, sister);
+        assertEquals("Marianne", me.getSiblings().get(0).getName());
+        assertEquals(1, me.getSiblings().size());
+    }
+
+    @Test
+    void addParents() {
+        me.addParents(father, mother, me);
+        assertEquals(father, me.getFather()); // manier 1
+        assertEquals("Johanna", me.getMother().getName()); // manier 2
+    }
+
+    @Test
+    void addPartnerTest() {
+        me.addPartner(me, wife);
+        assertEquals(me.getPartner(), wife);
+        assertEquals(wife.getPartner(), me);
+    }
+
+    // Deze lukt me niet... Ergens mis ik iets, maar ik kom er maar niet achter.
+    @Test
+    void getGrandchildrenTest() {
+        // Arrange
+        Person grandparent = new Person("Johanna", "Bonsma", 'V', 68); // Mijn moeder
+        Person parent = new Person("Marianne", "Bosma", 'V', 40); // Mijn zus, gekozen omdat zij twee kinderen heeft, zodat je de lijst kan vullen met meer dan 1 item.
+        Person child1 = new Person("Eva", "Johanna", "Veldhuizen", 'V', 8); // Oudste dochter van mijn zus
+        Person child2 = new Person("Rosa", "Tooske", "Veldhuizen", 'V', 5); // Jongste dochter van mijn zus
+
+        grandparent.getChildren().add(parent); // Verbindt mijn moeder en zus
```

Comment:
Dit werkt niet maar het zit een beetje ingewikkeld in elkaar waardoor het lastig is te zien waarom niet. In het kort:
-Je probeert personen toe te voegen door middel van de add() methode. Dit is niet een methode van de Person class, maar een methode van de generieke List<>...
-...en een List<> is wat je terugkrijgt uit de getChildren() functie. Je haalt dus een lijst op van kinderen en voegt daar een Person aan toe, maar slaat deze lijst nergens op. Deze lijst is geen property/eigenschap van een van deze personen (zoals de List<Person> Children) maar staat op zich. Het zou ook op deze manier geschreven kunnen zijn, met hetzelfde effect:
```
List<Person> wordtNietGebruikt = grandparent.getChildren(); // <= dit gebeurt in de eerste helft van de regel
wordtNietGebruikt.add(parent); // <= hier voeg je de Person parent aan de nieuwe, op zichzelf staande lijst toe maar niet aan de Children van grandparent
```

Om dit op te lossen zul je de addParents() of setMother() methode moeten toepassen, of de addChildren(), zodat de Persons aan de Children en/of Mother properties van de "eigen persoon" worden toegevoegd. Bijvoorbeeld:
```
List<Person> parentsList = new ArrayList<>(parent);
grandparent.addChildren(parentsList);
List<Person> childrenList = new ArrayList<>(child1, child2);
parent.addChildren(childrenList);
```

Reply:
Dank voor de uitleg! Zoals eerder genoemd, heb ik hierbij nog wat hulp gekregen in de laatste huiswerkklas. Hoe de methode er nu uitziet, is vergelijkbaar met wat jij hierboven schetst, en het werkt!